src/main/java/com/google/sps/data/PlaceReviews.java

```diff
@@ -0,0 +1,138 @@
+package com.google.sps.data;
+
+import com.google.sps.data.Comment;
+import com.google.sps.data.RatingHistory;
+import java.util.*;
+
+/** 
+ * PlaceReviews object class
+ * The PlaceReviews object, used as the master object detailing a locations reviews and ratings
+ * Uses implementations of smaller Comment and RatingHistory class.
+ * Does not permit multiple reviews by one author for a given location.
+ * Attempts to avoid PlaceReviews objects without any ratings or reviews.
+ */
+public class PlaceReviews {
+
+  private final String place_id;
+  private List<Comment> reviews;
+  private double rating;
+  private Set<String> reviewers;
+  private String currUser;
+
+   /**
+    * Constructor
+    * This is the minimal, and avoided, constructor that makes an object without reviews.
+    */
+  public PlaceReviews(String place_id) {
+    this.place_id = place_id;
+    this.reviews = new ArrayList<Comment>();
+    this.reviewers = new HashSet<>();
+  }
+
+  /** 
+   * Constructor
+   * This constructor provides an initial rating and review.
+   */
+  public PlaceReviews(String place_id, Comment firstReview, double initialRating) {
+    this.reviews = new ArrayList<Comment>();
+    this.place_id = place_id;
+    this.reviews.add(firstReview);
+    this.rating = initialRating;
+    this.reviewers = new HashSet<>();
+  }
+
+  /* Keeps track of who is currently logged in */
+  public void setCurrentUser(String person){
+    this.currUser = person;
+  }
+
+  /** 
+   * Updater method
+   * Adds a review to the existing PlaceReviews list of reviews
+   */
+  public void addReview(Comment review) {
+    if (this.reviewedBy(review.getAuthor())) {
+      replaceReview(review);
+    } else {
+      this.reviews.add(review);
+    }
+  }
+
+  /**
+   * Updater method
+   * Replaces previous review with the new review for a given author
+   */
+   private void replaceReview(Comment review) {
+    Comment prevReview = getPrevReview(review.getAuthor());
+    prevReview.updateComment(review);
+   } 
+
+  /**
+   * Accessor Method
+   * Searches the existing Array of comments to find the other by the same author
+   * Assumes that there was a previous post by author
+   * @return Comment the previous comment
+   *                  TODO: Implement with Binary Search
+   */
+   private Comment getPrevReview(String author) {
+    Collections.sort(this.reviews, Comment.ORDER_BY_AUTHOR);
+     for (Comment cur : this.reviews) {
```

Comment:
Indent.

Why is sorting needed here before a linear search?

Reply:
Not sure, will loop in @dterr

Reply:
I wrote in the sort by author as an attempt to implement a binary search, which I ended up not implementing. This is poor style for runtime and I will change it.

Reply:
#99

Reply:
Awesome - thanks for quickly fixing that!

Reply:
do we still need the todo or as this been resolved?